[BUG] LM Studio structured output fails: synthetic reasoning marker not stripped from content

## Summary

The `lmstudio` structured-output integration tests fail deterministically on `main`. They are the only genuine any-llm code bug in the current red integration suite (the other failures are provider account/routing problems, tracked separately).

Failing tests (both fail on every attempt, 6/6 retries):

- `tests/integration/test_response_format.py::test_response_format[lmstudio]`
- `tests/integration/test_response_format.py::test_response_format_dataclass[lmstudio]`

Seen in the two most recent post-merge Integration Tests runs:
- https://github.com/mozilla-ai/any-llm/actions/runs/29012376803
- https://github.com/mozilla-ai/any-llm/actions/runs/29010060133

## Root cause

The CI model is a reasoning model (`qwen/qwen3-1.7b`, see `tests/conftest.py`). When `response_format` is requested, LM Studio's native SDK returns the reasoning trace **inline in `PredictionResult.content`**, delimited from the answer by a synthetic marker of the form:

```
__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_<hex>__
```

Actual `content` returned (abbreviated):

```
Okay, the user is asking for the capital of France. Let me think about this. ... ris.
__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_f4e9a8d2c6b14d0c9e5f3a7b8c1d2e6a__{
  "city_name": "Paris"
}
```

`src/any_llm/providers/lmstudio/utils.py::_split_reasoning_from_content` only strips reasoning wrapped in `<think>...</think>` tags. It does not recognize the `__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_...__` delimiter, so the entire string (reasoning prose + JSON) is passed through as `content`.

`any_llm.py:645` then calls `parse_json_content(response_format, content)` (`src/any_llm/utils/structured_output.py`), which does `model_validate_json` / `TypeAdapter(...).validate_json` on a string that starts with prose, raising:

```
pydantic_core._pydantic_core.ValidationError: 1 validation error for ResponseFormat
  Invalid JSON: expected value at line 1 column 1 [type=json_invalid, ...]
```

The non-reasoning local providers (`llamacpp`, `ollama`) pass the same tests, confirming this is specific to how the LM Studio native SDK surfaces reasoning content.

## Stack trace

```
src/any_llm/any_llm.py:645: in acompletion
    choice.message.parsed = parse_json_content(response_format, choice.message.content)
src/any_llm/utils/structured_output.py:48: in parse_json_content
    return response_format.model_validate_json(content)
E   pydantic_core._pydantic_core.ValidationError: 1 validation error for ResponseFormat
E     Invalid JSON: expected value at line 1 column 1 [type=json_invalid, input_value='Okay, the user is asking..."city_name": "Paris"\n}', input_type=str]
```

## Suggested direction (not yet implemented)

Extend `_split_reasoning_from_content` (or the LM Studio response conversion) to also split on the `__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_<hex>__` marker, routing the prefix to `reasoning.content` and keeping only the trailing JSON as `content`. Add a unit test covering a `content` payload that uses the synthetic marker instead of `<think>` tags.

## Environment

- any-llm: `main` @ `48b60a9`
- Python: 3.13 (CI)
- Model: `qwen/qwen3-1.7b` via LM Studio native SDK

--- a/src/any_llm/providers/lmstudio/utils.py
+++ b/src/any_llm/providers/lmstudio/utils.py
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import re
 import time
 import uuid
 from typing import TYPE_CHECKING, Any, Literal, cast
@@ -64,16 +65,27 @@ def _usage_from_stats(stats: LlmPredictionStats | None) -> CompletionUsage:
     )
 
 
-def _split_reasoning_from_content(content: str) -> tuple[str, str | None]:
-    """Extract reasoning wrapped in <think></think> tags from the content (if present).
+_SYNTHETIC_REASONING_END = re.compile(r"__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_[0-9a-f]+__")
 
-    Any text before the opening tag or after the closing tag is preserved as content,
-    so a model that emits a preface before its reasoning block does not lose output.
+
+def _split_reasoning_from_content(content: str) -> tuple[str, str | None]:
+    """Extract reasoning from the content (if present).
+
+    Handles the two shapes LM Studio surfaces depending on the model:
+    - reasoning wrapped in <think></think> tags, where any text before the opening tag or
+      after the closing tag is preserved as content
+    - a reasoning prefix terminated by a synthetic
+      __LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_<hex>__ marker, emitted by reasoning
+      models when a response_format is requested; the prefix is reasoning and the trailing text
+      (the structured JSON answer) is the content
     """
     if "<think>" in content and "</think>" in content:
         before, after_open = content.split("<think>", 1)
         reasoning, after_close = after_open.split("</think>", 1)
         return before + after_close, reasoning
+    match = _SYNTHETIC_REASONING_END.search(content)
+    if match:
+        return content[match.end() :], content[: match.start()]
     return content, None
 
 

--- a/src/any_llm/providers/lmstudio/utils.py
+++ b/src/any_llm/providers/lmstudio/utils.py
@@ -65,7 +65,7 @@ def _usage_from_stats(stats: LlmPredictionStats | None) -> CompletionUsage:
     )
 
 
-_SYNTHETIC_REASONING_END = re.compile(r"__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_[0-9a-f]+__")
+_SYNTHETIC_REASONING_END = re.compile(r"__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_[0-9a-fA-F]+__")
 
 
 def _split_reasoning_from_content(content: str) -> tuple[str, str | None]:

--- a/tests/unit/providers/test_lmstudio_provider.py
+++ b/tests/unit/providers/test_lmstudio_provider.py
@@ -390,6 +390,19 @@ def test_map_stop_reason(stop_reason: str | None, expected: str) -> None:
         ("<think>because</think>final", "final", "because"),
         ("preface <think>because</think> final", "preface  final", "because"),
         ("no tags here", "no tags here", None),
+        # LM Studio's runtime ends an inline reasoning trace with a fixed synthetic marker
+        # when response_format is set (see #1167). The value below is the real marker observed
+        # in CI; the split must work for any hex nonce, so the next case uses a different one.
+        (
+            'Paris is the capital.\n__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_f4e9a8d2c6b14d0c9e5f3a7b8c1d2e6a__{\n  "city_name": "Paris"\n}',
+            '{\n  "city_name": "Paris"\n}',
+            "Paris is the capital.\n",
+        ),
+        (
+            "reasoning...__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_0123456789abcdef__{}",
+            "{}",
+            "reasoning...",
+        ),
     ],
 )
 def test_split_reasoning_from_content(content: str, expected_content: str, expected_reasoning: str | None) -> None:

--- a/tests/unit/providers/test_lmstudio_provider.py
+++ b/tests/unit/providers/test_lmstudio_provider.py
@@ -392,7 +392,8 @@ def test_map_stop_reason(stop_reason: str | None, expected: str) -> None:
         ("no tags here", "no tags here", None),
         # LM Studio's runtime ends an inline reasoning trace with a fixed synthetic marker
         # when response_format is set (see #1167). The value below is the real marker observed
-        # in CI; the split must work for any hex nonce, so the next case uses a different one.
+        # in CI; the split must work for any hex nonce, so the next cases use a different
+        # lowercase nonce and an uppercase one.
         (
             'Paris is the capital.\n__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_f4e9a8d2c6b14d0c9e5f3a7b8c1d2e6a__{\n  "city_name": "Paris"\n}',
             '{\n  "city_name": "Paris"\n}',
@@ -400,6 +401,11 @@ def test_map_stop_reason(stop_reason: str | None, expected: str) -> None:
         ),
         (
             "reasoning...__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_0123456789abcdef__{}",
+            "{}",
+            "reasoning...",
+        ),
+        (
+            "reasoning...__LM_STUDIO_INTERNAL_LSEP_SYNTHETIC_REASONING_END_DEADBEEF01234567__{}",
             "{}",
             "reasoning...",
         ),

